cli/compose/convert/volume.go

```diff
@@ -61,17 +61,18 @@ func convertVolumeToMount(
 		return result, errors.Errorf("undefined volume %q", volume.Source)
 	}
 
+	if stackVolume.External.External && stackVolume.External.Name != "" {
+		return result, errors.Errorf("cannot specify external volume name %q", stackVolume.External.Name)
+	}
 	result.Source = namespace.Scope(volume.Source)
 	result.VolumeOptions = &mount.VolumeOptions{}
 
 	if volume.Volume != nil {
 		result.VolumeOptions.NoCopy = volume.Volume.NoCopy
 	}
 
-	// External named volumes
-	if stackVolume.External.External {
-		result.Source = stackVolume.External.Name
-		return result, nil
```

Comment:
We still need this block to support `External.External`.  

We could also add a deprecation message in the loader